kolibri/core/assets/src/vue/content-icon/index.vue

```diff
@@ -0,0 +1,119 @@
+<template>
+
+  <div>
+    <svg
+      viewbox="0 0 32 32"
+      class="content-icon"
+      :title="altText">
+      <circle
+        cy="16"
+        cx="16"
+        r="16"
+        class="progress-circle"
+        :style="{ 'stroke-dasharray': progressPercent + ' 100' }">
+      </circle>
+      <circle
+        cy="16"
+        cx="16"
+        r="13.5"
+        class="inner-circle">
+      </circle>
+      <svg v-if="thisIs('audio')" src="./content-icons/audio.svg" class="content-type-icon"></svg>
+      <svg v-if="thisIs('document')" src="./content-icons/document.svg" class="content-type-icon"></svg>
+      <svg v-if="thisIs('exercise')" src="./content-icons/exercise.svg" class="content-type-icon"></svg>
+      <svg v-if="thisIs('video')" src="./content-icons/video.svg" class="content-type-icon"></svg>
+    </svg>
+  </div>
+
+</template>
+
+
+<script>
+
+  const ContentKinds = require('core-constants').ContentKinds;
+
+  module.exports = {
+    $trNameSpace: 'learn',
+    $trs: {
+      complete: 'complete',
+      partial: 'partial',
+      unstarted: 'unstarted',
+      audio: 'audio',
+      document: 'document',
+      video: 'video',
+    },
+    props: {
+      ispageicon: {
+        type: Boolean,
+        default: false,
+      },
+      size: {
+        type: Number,
+        default: 30,
+      },
+      progress: {
+        type: Number,
+        default: 0.0,
+        validator(value) {
+          return (value >= 0.0) && (value <= 1.0);
+        },
+      },
+      kind: {
+        type: String,
+        required: true,
+        validator(value) {
+          for (const contentKind in ContentKinds) {
+            if (ContentKinds[contentKind] === value) {
+              return true;
+            }
+          }
+          return false;
+        },
```

Comment:
Ya, I couldn't find a more concise since we are looking at the values of the object.